WordPress/gutenberg
# Edit Site: Add theme exporter.

Closes #19260 

## Description

This PR adds a theme exporter button to the site editor header's "More" dropdown.

The exporter uses @melchoyce's [Johannes theme](https://github.com/melchoyce/johannes) as the base for the theme zip file and adds the site editor's templates and template parts.

## How to test this?

Try exporting templates/parts with different modifications and verify that the exported files match what you have on the editor.

## Screenshots

<img width="382" alt="Screen Shot 2020-04-28 at 12 52 18 PM" src="https://user-images.githubusercontent.com/19157096/80531708-e673ec80-894f-11ea-82ef-f1d96e285273.png">

<img width="839" alt="Screen Shot 2020-04-28 at 12 52 30 PM" src="https://user-images.githubusercontent.com/19157096/80531719-eaa00a00-894f-11ea-9c1f-abcc505c335b.png">

## Types of Changes

*New Feature:* The site editor now has a theme exporter button for exporting customized templates and template parts in a theme zip file.

## Checklist:
